File: 99_bottles.py
numberRequested = int(input("How many bottles? "))

def bottle_song(numberOfBottles, count):
	
	if numberOfBottles == 0:
		print("No more bottles of beer on the wall, no more bottles of beer.")
		print(f'Go to the store and buy some more, {count} bottles of beer on the wall.')
	elif numberOfBottles == 1:
		print(f'{numberOfBottles} bottle of beer on the wall, {numberOfBottles} bottle of beer.')
		print("Take one down and pass it around, no more bottles of beer on the wall.")
		bottle_song(numberOfBottles - 1, count + 1)
	elif numberOfBottles == 2:
		print(f'{numberOfBottles} bottle of beer on the wall, {numberOfBottles} bottle of beer.')
		print(f'Take one down and pass it around, {numberOfBottles-1} bottle of beer on the wall.')
		bottle_song(numberOfBottles - 1, count + 1)
	else:
		print(f'{numberOfBottles} bottles of beer on the wall, {numberOfBottles} bottles of beer.')
		print(f'Take one down and pass it around, {numberOfBottles - 1} bottles of beer on the wall.')
		bottle_song(numberOfBottles - 1, count + 1)

bottle_song(numberRequested, 0)

File: 99_bottles.js
const readline = require("readline");
const rl = readline.Interface({
  input: process.stdin,
  output: process.stdout
});

function bottleSong() {
  rl.question("How many Bottles?", function(numberOfBottles){
    let lastbottle = false;
    let line1 = "";
    let line2 = "";
    for (let i = numberOfBottles; i >= 0; i--) {
      let j = i - 1;
      let remainingBottles = (j == 0 ? "no more" : j);
      let startBottlePlural = ((numberOfBottles == 1) ? "bottle" : "bottles");
      let currentBottlePlural = ((i == 1) ? "bottle" : "bottles");
      let nextBottlePlural = ((j == 1) ? "bottle" : "bottles");
      
      if (lastbottle) {
        line1 = "No more bottles of beer on the wall, no more bottles of beer."
        line2 = `Go to the store and buy some more, ${numberOfBottles} ${startBottlePlural} of beer on the wall.`
        lastbottle = false;
      } else {
        line1 = `${i} ${currentBottlePlural} of beer on the wall, ${i} ${currentBottlePlural} of beer.`
        line2 = `Take one down and pass it around, ${remainingBottles} ${nextBottlePlural} of beer on the wall.`
        if (i == 1) {lastbottle = true}
    }
    console.log(line1)
    console.log(line2)
  }
  process.exit(0);
  })
  
};

bottleSong();
